src/layout/Main/index.tsx

```diff
@@ -137,21 +144,51 @@ function MainLayout({ children }: PropsWithChildren) {
   const globalElements = useAtomValue(globalElementsAtom);
   const { walletAddress } = useConnectedWallet();
   const connectWalletModal = useConnectWalletModal();
-  const [searchParam, setSearchParam] = useSearchParamState<string>(
-    CHAIN_NAME_SEARCH_PARAM,
-    DefaultChainName,
-  );
+  const [searchParams, setSearchParams] = useSearchParams();
 
-  const name = useMemo(() => {
-    if (wallet.status === WalletStatus.WALLET_CONNECTED) {
-      return wallet.network.name === "testnet" ? "xplatestnet" : "xpla";
-    }
-    return searchParam ?? DefaultChainName;
-  }, [searchParam, wallet.network.name, wallet.status]);
+  const handleModalClose = useCallback(() => {
+    const newParams = new URLSearchParams(searchParams);
+    newParams.set(CHAIN_NAME_SEARCH_PARAM, DefaultChainName);
+    setSearchParams(newParams);
+  }, [searchParams, setSearchParams]);
+
+  const { isOpen, open } = useInvalidPathModal({
+    onReturnClick: handleModalClose,
+  });
+
+  const { paramChainName, isValidChain } = useMemo(() => {
+    const paramValue = searchParams.get(CHAIN_NAME_SEARCH_PARAM);
+    return { paramChainName: paramValue, ...getValidChain(paramValue || "") };
+  }, [searchParams]);
 
   useEffect(() => {
-    setSearchParam(name);
-  }, [name, setSearchParam]);
+    open(!!(paramChainName && !isValidChain));
+  }, [isOpen, isValidChain, open, paramChainName, searchParams]);
```

Comment:
```suggestion
  }, [isValidChain, open, paramChainName]);
```

should we update this dependencies array like this?

Reply:
[8a6b329](https://github.com/dezswap/dezswap-web-app/pull/248/commits/8a6b32933506340e70168268bd429562d46e319a) is updated